server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1405,7 +1405,8 @@ public ReplicationCheckpoint getLatestReplicationCheckpoint() {
         if (indexSettings.isSegRepEnabled() == false) {
             return null;
         }
-        if (getEngineOrNull() == null) {
+        final IndexMetadata indexMetadata = indexSettings.getIndexMetadata();
+        if (getEngineOrNull() == null || (indexMetadata != null && indexMetadata.getState() == IndexMetadata.State.CLOSE)) {
```

Comment:
Is it possible for IndexMetadata to be null here?  Looks like from the IndexSettings [constructor](https://github.com/opensearch-project/OpenSearch/blob/3d90d35f6441e7e238fd3e6c5bf6e5c9e3e25eab/server/src/main/java/org/opensearch/index/IndexSettings.java#L700) or `updateIndexMetadata ` this would throw a NPE.

Reply:
Thanks for catching this. Updated.